src/test/java/app/controllers/HomeControllerTest.java

```diff
@@ -49,15 +54,29 @@ public void setUp() {
     }
 
     @Test
-    @WithMockUser(username = "username", roles={"STUDENT"})
+    @WithMockUser(username = USERNAME, roles={"STUDENT"})
     public void getStudentTemplate() {
-        projectRepository.save(project);
+        authenticateStudent();
         try {
-            mockMvc.perform(get("/student/"+student.getId()))
+            mockMvc.perform(get("/studentMenu/"))
                     .andExpect(status().isOk())
                     .andExpect(content().string(Matchers.containsString("<title>Fourth Year Project - Student</title>")));
         } catch (Exception e) {
             fail(e.getMessage());
         }
     }
+
+    private void authenticateStudent() {
```

Comment:
Check out this method if you want to do controller tests with an authenticated user